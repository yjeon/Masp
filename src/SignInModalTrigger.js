import React, { Component } from 'react';
import Modal from 'react-modal';


class SignInModalTrigger extends Component {
  constructor() {
    super();
    this.state = {
      open: false
    };
    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  openModal () {
    this.setState({open: true});
  }

  closeModal () {
    this.setState({open: false});
  }

  render () {
    return (
      <div>
        <a className="navbar-items" onClick={this.openModal} href="#">Sign in</a>
        <Modal className="sign-in-modal" isOpen={this.state.open}>
          <h1 className="modal-title">Sign in</h1>
          <div className="modal-inner-form">
            <input className="modal-input" id="modal-sign-in-mail-input" type="email" placeholder="Enter your email here"/>
            <br />
            <br />
            <input className="modal-input" id="modal-sign-in-password-input" type="password" placeholder="Enter your password here" pattern="(?=.*[A-Z]).{6,}"/>
            <br />
            <br />
            <button className="modal-button">Submit</button>
            <br />
            <br />
            <button className="modal-button" onClick={this.closeModal}>Close</button>
          </div>
        </Modal>
      </div>
    );
  }
}

export default SignInModalTrigger;
