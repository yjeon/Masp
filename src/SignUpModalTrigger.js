import React, { Component } from 'react';
import Modal from 'react-modal';
import firebase from 'firebase';

class SignUpModalTrigger extends Component {
  constructor() {
    super();
    this.state = {
      open: false
    };
    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
    this.submit = this.submit.bind(this);
  }

  openModal () {
    this.setState({open: true});
  }

  closeModal () {
    this.setState({open: false});
  }

  //this function sends data to firebase server
  submit() {
    const email  = document.getElementById('modal-sign-up-mail-input').value;
    const pass   = document.getElementById('modal-sign-up-password-input').value;
    const promise = firebase.auth().createUserWithEmailAndPassword(email, pass);
    promise.then(e => window.location.href="index.html");
    promise.catch(e => console.log(e.message));

    firebase.auth().onAuthStateChanged(firebaseUser => {
      if(firebaseUser) {
        this.setState({open: false});
        console.log("hi");
      }
    });
  }

  render () {
    return (
      <div>
        <a className="navbar-items" onClick={this.openModal} href="#">Sign up</a>
        <Modal className="sign-up-modal" isOpen={this.state.open}>
          <h1 className="modal-title">Sign Up</h1>
          <div className="modal-inner-form">
            <input className="modal-input" id="modal-sign-up-mail-input" type="email" placeholder="Enter your email here"/>
            <br />
            <br />
            <input className="modal-input" id="modal-sign-up-password-input" type="password" placeholder="Enter your password here" pattern="(?=.*[A-Z]).{6,}"/>
            <br />
            <br />
        <button className="modal-button" id="modal-sign-up-submit-button" onClick={this.submit}>Submit</button>
            <br />
            <br />
            <button className="modal-button" onClick={this.closeModal}>Close</button>
          </div>
        </Modal>
      </div>
    );
  }
}

export default SignUpModalTrigger;
