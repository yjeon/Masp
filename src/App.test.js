import SignInModalTrigger from './SignInModalTrigger';
import SignUpModalTrigger from './SignUpModalTrigger';
var firebase = jest.fn(); //a auto-generated mock for module firebase

/*
 * beforeEach(*
 * method is gotten from https://github.com/facebook/jest/issues/1353
 * as a work around to fix react-dom and react-test-renderer to call twice,
 * resulting invariant violation.
 * this issue will be fixed in jest v.15.4
 */

beforeEach(() => jest.resetModules());

test('Sign-in-modal opened correctly (using react-dom)', () => {
  const React = require('react');
  const enzyme = require('enzyme');
  // write your test with enzyme
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SignInModalTrigger />, div);
  });
});

test('Sign-in-modal opened correctly (using react-test-renderer)', () => {
  const React = require('react');
  const renderer = require('react-test-renderer');
  // write your test with react-test-renderer
  const component = renderer.create(
    <SignInModalTrigger></SignInModalTrigger>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Sign-in-modal opened correctly and submit working correctly', () => {
    const React = require('react');
    const renderer = require('react-test-renderer');
    // write your test with react-test-renderer
    const component = renderer.create(
            <SignInModalTrigger></SignInModalTrigger>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.submit;
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Sign-in-modal opened correctly and closed correctly', () => {
    const React = require('react');
    const renderer = require('react-test-renderer');
    // write your test with react-test-renderer
    const component = renderer.create(
            <SignInModalTrigger></SignInModalTrigger>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.closeModal;
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Sign-up-modal opened correctly (using react-dom)', () => {
    const React = require('react');
    const enzyme = require('enzyme');
    // write your test with enzyme
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<SignUpModalTrigger />, div);
    });
});

test('Sign-up-modal opened correctly (using react-test-renderer)', () => {
    const React = require('react');
    const renderer = require('react-test-renderer');
    // write your test with react-test-renderer
    const component = renderer.create(
            <SignUpModalTrigger></SignUpModalTrigger>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Sign-up-modal opened correctly and submit working correctly', () => {
    const React = require('react');
    const renderer = require('react-test-renderer');
    // write your test with react-test-renderer
    const component = renderer.create(
            <SignUpModalTrigger></SignUpModalTrigger>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.submit;
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Sign-up-modal opened correctly and closed correctly', () => {
    const React = require('react');
    const renderer = require('react-test-renderer');
    // write your test with react-test-renderer
    const component = renderer.create(
            <SignUpModalTrigger></SignUpModalTrigger>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.closeModal;
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
