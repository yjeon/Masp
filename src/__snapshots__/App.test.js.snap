exports[`test Sign-in-modal opened correctly (using react-test-renderer) 1`] = `
<div>
  <a
    className="navbar-items"
    href="#"
    id="sign-in-open-modal"
    onClick={[Function bound openModal]}>
    Sign in
  </a>
  <noscript />
</div>
`;

exports[`test Sign-in-modal opened correctly and closed correctly 1`] = `
<div>
  <a
    className="navbar-items"
    href="#"
    id="sign-in-open-modal"
    onClick={[Function bound openModal]}>
    Sign in
  </a>
  <noscript />
</div>
`;

exports[`test Sign-in-modal opened correctly and closed correctly 2`] = `
<div>
  <a
    className="navbar-items"
    href="#"
    id="sign-in-open-modal"
    onClick={[Function bound openModal]}>
    Sign in
  </a>
  <noscript />
</div>
`;

exports[`test Sign-in-modal opened correctly and submit working correctly 1`] = `
<div>
  <a
    className="navbar-items"
    href="#"
    id="sign-in-open-modal"
    onClick={[Function bound openModal]}>
    Sign in
  </a>
  <noscript />
</div>
`;

exports[`test Sign-in-modal opened correctly and submit working correctly 2`] = `
<div>
  <a
    className="navbar-items"
    href="#"
    id="sign-in-open-modal"
    onClick={[Function bound openModal]}>
    Sign in
  </a>
  <noscript />
</div>
`;

exports[`test Sign-up-modal opened correctly (using react-test-renderer) 1`] = `
<div>
  <a
    className="navbar-items"
    href="#"
    id="sign-up-open-modal"
    onClick={[Function bound openModal]}>
    Sign up
  </a>
  <noscript />
</div>
`;

exports[`test Sign-up-modal opened correctly and closed correctly 1`] = `
<div>
  <a
    className="navbar-items"
    href="#"
    id="sign-up-open-modal"
    onClick={[Function bound openModal]}>
    Sign up
  </a>
  <noscript />
</div>
`;

exports[`test Sign-up-modal opened correctly and closed correctly 2`] = `
<div>
  <a
    className="navbar-items"
    href="#"
    id="sign-up-open-modal"
    onClick={[Function bound openModal]}>
    Sign up
  </a>
  <noscript />
</div>
`;

exports[`test Sign-up-modal opened correctly and submit working correctly 1`] = `
<div>
  <a
    className="navbar-items"
    href="#"
    id="sign-up-open-modal"
    onClick={[Function bound openModal]}>
    Sign up
  </a>
  <noscript />
</div>
`;

exports[`test Sign-up-modal opened correctly and submit working correctly 2`] = `
<div>
  <a
    className="navbar-items"
    href="#"
    id="sign-up-open-modal"
    onClick={[Function bound openModal]}>
    Sign up
  </a>
  <noscript />
</div>
`;
